skimage/draw/tests/test_draw.py

```diff
@@ -891,3 +892,102 @@ def test_rectangle_extent():
+    assert_array_equal(img, expected)
+
+    expected = np.array([[0, 0, 0, 0, 0, 0],
+                         [0, 0, 1, 1, 1, 0],
+                         [0, 0, 1, 0, 1, 0],
+                         [0, 0, 1, 1, 1, 0],
+                         [0, 0, 0, 0, 0, 0]], dtype=np.uint8)
+    img = np.zeros(expected.shape, dtype=np.uint8)
+    start = (2, 3)
+    end = (2, 3)
+    rr, cc = rectangle_perimeter(start, end=end, shape=img.shape)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
+
+    expected = np.array([[0, 0, 0, 0, 0, 0],
+                         [0, 0, 1, 1, 1, 1],
+                         [0, 0, 1, 0, 0, 1],
+                         [0, 0, 1, 1, 1, 1],
+                         [0, 0, 0, 0, 0, 0]], dtype=np.uint8)
+    img = np.zeros(expected.shape, dtype=np.uint8)
+    start = (2, 3)
+    end = (2, 4)
+    rr, cc = rectangle_perimeter(start, end=end, shape=img.shape)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
+
+
+def test_rectangle_perimiter_clip():
+    expected = np.array([[0, 0, 0, 0, 0],
+                         [0, 1, 1, 1, 1],
+                         [0, 1, 0, 0, 0],
+                         [0, 1, 0, 0, 0],
+                         [0, 1, 0, 0, 0]], dtype=np.uint8)
+    img = np.zeros((5, 5), dtype=np.uint8)
+    start = (2, 2)
+    extent = (10, 10)
+    rr, cc = rectangle_perimeter(start, extent=extent, shape=img.shape,
+                                 clip=False)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
+
+    expected = np.array([[0, 0, 0, 0, 0],
+                         [0, 1, 1, 1, 1],
+                         [0, 1, 0, 0, 1],
+                         [0, 1, 0, 0, 1],
+                         [0, 1, 1, 1, 1]], dtype=np.uint8)
+    img = np.zeros((5, 5), dtype=np.uint8)
+    start = (2, 2)
+    extent = (10, 10)
+    rr, cc = rectangle_perimeter(start, extent=extent, shape=img.shape,
+                                 clip=True)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
+
+    expected = np.array([[0, 0, 0, 1, 0],
+                         [0, 0, 0, 1, 0],
+                         [0, 0, 0, 1, 0],
+                         [1, 1, 1, 1, 0],
+                         [0, 0, 0, 0, 0]], dtype=np.uint8)
+    img = np.zeros((5, 5), dtype=np.uint8)
+    start = (-5, -5)
+    end = (2, 2)
+    rr, cc = rectangle_perimeter(start, end=end, shape=img.shape,
+                                 clip=False)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
+
+    expected = np.array([[1, 1, 1, 1, 0],
+                         [1, 0, 0, 1, 0],
+                         [1, 0, 0, 1, 0],
+                         [1, 1, 1, 1, 0],
+                         [0, 0, 0, 0, 0]], dtype=np.uint8)
+    img = np.zeros((5, 5), dtype=np.uint8)
+    start = (-5, -5)
+    end = (2, 2)
+    rr, cc = rectangle_perimeter(start, end=end, shape=img.shape,
+                                 clip=True)
+    img[rr, cc] = 1
+    assert_array_equal(img, expected)
```

Comment:
I'd add a test with upper right start and lower left end / (+, -) extent.

Also, a very simple 3D test would be nice.